test/e2e/storage/testsuites/base.go

```diff
@@ -241,6 +242,16 @@ func createGenericVolumeTestResource(driver TestDriver, config *PerTestConfig, p
 	return &r
 }
 
+func createVolumeSource(pvcName string, readOnly bool) *v1.VolumeSource {
```

Comment:
I guess from a storage perspective, it made sense to split this one out? If your only goal was to de-dupe the instantiation of the PVC VolumeSource, then I think it was small enough to have twice.